### PR TITLE
Fix Website to load urls from data file instead of hardcoded url

### PR DESCRIPTION
**Pull request by rajmbcoderx**
- Use url.yaml to load the URLs for processing status and config
- Update website docker image to use extender hugo version to enable npm installation, this enables proper rendering of react pages in the website
- Merged hugo-react-build-bug to this branch

Fixes https://github.com/filetrust/cdr-plugin-folder-to-folder/issues/420 website fails to build on docker and docker-compose

Testing this locally:
1. Copy urls-dev.yaml content to urls.dev
2. Execute docker-compose up --build
3. Goto http://localhost:1313/status/server_config/ and validate the config is loaded properly

We already have a ticket to ensure that we add tests for the UI https://github.com/filetrust/cdr-plugin-folder-to-folder/issues/381

### CODE REVIEW

File: website/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
-FROM        klakegg/hugo:ubuntu
+FROM        klakegg/hugo:ext-ubuntu
 RUN         apt update
 RUN         apt install -y git
 RUN         git clone https://github.com/matcornic/hugo-theme-learn.git
 WORKDIR     site
-ENTRYPOINT  hugo serve
+ENTRYPOINT  npm install && hugo serve
```
Comment (DinisCruz) [inline website/Dockerfile:6]: npm install should not be done in the same line as the ENTRYPOINT

Can you merge this branch into this PR https://github.com/filetrust/cdr-plugin-folder-to-folder/tree/fix-website-with-react

if hugo:ext-ubuntu comes with node install then let's use that (on the version in the branch above we were installing node)

File: website/site/layouts/shortcodes/status/server_config.html [hunk 2]
```diff
@@ -5,7 +5,7 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     {{ with resources.Get "js/HighlightTexts.jsx" }}
-    {{ $script := . | js.Build (dict "params" (dict "url" "http://localhost:8000/configuration/config/"))}}
+    {{ $script := . | js.Build (dict "params" (dict "url"  (printf "%s/configuration/config/" $.Site.Data.urls.api)))}}
     <script src="{{ $script.RelPermalink }}" defer></script>
```
Comment (DinisCruz) [inline website/site/layouts/shortcodes/status/server_config.html:9]: Good changes

one worry I have here is that we don't have a test for these changes

how do we know they worked?

**Review by DinisCruz — CHANGES_REQUESTED** [review]

**Comment by DinisCruz** [comment]
todo:
 - merge `fix-website-with-react` branch into this code
 - add test to check for URLs in live server (these are integration tests with hugo running locally)
 - add to the description of this issue the ids/links to the issues that this PR addresses

File: website/Dockerfile (hunk 1 above)
Comment (rajmbcoderx) [reply website/Dockerfile:6]: @DinisCruz I had to do npm install here as we are mounting the site content as volumes and hence I had to execute npm install in entrypoint as that is the only place where I can ensure that all the content is mounted, I could have ensured that we have all the content copied to docker image while building, but we loose the capability of dynamic update to the content.

The branch you mentioned is the current branch I created the PR on, is there any other branch/PR that you are referring to?

File: website/site/layouts/shortcodes/status/server_config.html (hunk 2 above)
Comment (rajmbcoderx) [reply website/site/layouts/shortcodes/status/server_config.html:9]: @DinisCruz Yes, I currently don't have a way to test this. We would need to enable UI testing. @nishantswarup Do you have any suggestions on how to write tests for these changes?

File: website/Dockerfile (hunk 1 above)
Comment (rajmbcoderx) [reply website/Dockerfile]: @DinisCruz Merged hugo-react-build-bug and also ensured that npm install is not done in entrypoint

**Review by nishantswarup — APPROVED** [review]

**Pull request merged**